src/main/java/org/sagebionetworks/web/client/widget/entity/browse/EntityFinderV2ViewImpl.java

```diff
@@ -0,0 +1,271 @@
+
+	@Override
+	public void clear() {
+		clearError();
+		entityFinderContainer.clear();
+		presenter.clearSelectedEntities();
+	}
+
+	@Override
+	public void show() {
+		// show modal
+		modal.show();
+		presenter.renderComponent();
+		helpWidget.focus();
+	}
+
+	@Override
+	public void hide() {
+		this.jsniUtils.unmountComponentAtNode(entityFinderContainer.getElement());
+		modal.hide();
+	}
+
+	@Override
+	public Widget asWidget() {
+		return modal;
+	}
+
+	@Override
+	public void setErrorMessage(String errorMessage) {
+		clearError();
+		synAlert.showError(errorMessage);
+	}
+
+	@Override
+	public void clearError() {
+		synAlert.clearState();
+	}
+
+	@Override
+	public void setModalTitle(String modalTitle) {
+		this.modalTitle.setText(modalTitle);
+	}
+
+	@Override
+	public void setPromptCopy(String promptCopy) {
+		this.promptCopy.setText(promptCopy);
+	}
+
+	@Override
+	public void setHelpMarkdown(String helpMarkdown) {
+		this.helpWidget.setHelpMarkdown(helpMarkdown);
+	}
+
+	@Override
+	public void setConfirmButtonCopy(String confirmButtonCopy) {
+		this.okButton.setText(confirmButtonCopy);
+	}
+
+	public static class ReferenceJso extends JavaScriptObject {
+
+		protected ReferenceJso() {}
+
+		public final native String getTargetId() /*-{ return this.targetId }-*/;
+
+		public final native int getTargetVersionNumber() /*-{ return this.targetVersionNumber || -1 }-*/;
+
+	}
+
+	private Reference convertFromJso(ReferenceJso jso) {
+		Reference r = new Reference();
+		r.setTargetId(jso.getTargetId());
+		if (jso.getTargetVersionNumber() == -1) {
+			r.setTargetVersionNumber(null);
+		} else {
+			r.setTargetVersionNumber(Long.valueOf(jso.getTargetVersionNumber()));
+		}
+		return r;
+	}
+
+	private static native void _showEntityFinderReactComponent(Element el, String sessionToken, String projectId, String initialContainer, String initialScope, JsArrayString selectableTypes, JsArrayString visibleTypesInList, JsArrayString visibleTypesInTree, boolean showVersions, boolean multiSelect, String selectedCopy, boolean treeOnly, OnSelectCallback onSelectedCallback) /*-{
+		try {
+			var callback = function(selected) {
+				onSelectedCallback.@org.sagebionetworks.web.client.callback.OnSelectCallback::onSelect(Lcom/google/gwt/core/client/JsArray;)(selected)
+			};
+			var props = {
+				sessionToken: sessionToken,
+				projectId: projectId,
+				initialContainer: initialContainer,
+				initialScope: initialScope,
+				visibleTypesInList: visibleTypesInList,
+				selectableTypes: selectableTypes,
+				visibleTypesInTree: visibleTypesInTree,
+				selectMultiple: multiSelect,
+				onSelectedChange: callback,
+				showVersionSelection: showVersions,
+				selectedCopy: selectedCopy,
+				treeOnly: treeOnly,
+			};
+			$wnd.ReactDOM.render(
+				$wnd.React.createElement($wnd.SRC.SynapseComponents.EntityFinder, props, null),
+				el
+			);
+		} catch (err) {
+			console.error(err);
+		}
+	}-*/;
+
+	public static JsArrayString toJsArray(List<String> list) {
```

Comment:
This seems like it is of general use, and should be in a js utility class.  Perhaps convertFromJso and ReferenceJso too.